src/index.ts

```diff
@@ -416,7 +416,7 @@ abstract class Filter extends Plugin {
       return false;
     }
 
-    return (entry || this.input.lstatSync(relativePath)).isDirectory();
+    return entry ? entry.isDirectory() : this.input.existsSync(relativePath) && this.input.lstatSync(relativePath).isDirectory();
```

Comment:
This is going to do two stat operations (one for input.existsSync and one for input.lstatSync), but I think we can actually avoid that. Additionally, this is mixing the concepts of stat and lstat (`existsSync` does a stat and therefore follows symlinks, but then we are actually using `lstat().isDirectory()`).

```js
if (entry !== undefined) {
  return entry.isDirectory();
} else {
  try {
    const stat = this.input.lstatSync(relativePath);
    return stat.isDirectory();
  } catch (error) {
    // if we get any other error, we really don't know what is going on so we need to rethrow
    if (error.code === 'ENOENT') {
      return false;
    }
    throw error;
  }
}
```

(Apologies for typos here, typed it out on mobile)

Reply:
lol, see: https://github.com/dfreeman/broccoli-graphql-filter/pull/20/files